docs/side_quests/splitting_and_grouping.md

```diff
@@ -0,0 +1,1298 @@
+[id:sampleA, repeat:2, type:tumor, bam:sampleB_rep1_tumor.bam]
+[id:sampleB, repeat:1, type:tumor, bam:sampleC_rep1_tumor.bam]
+[id:sampleC, repeat:1, type:tumor, bam:sampleD_rep1_tumor.bam]
+```
+
+Note how we can only see the tumor samples in the output. This is because we removed the `view` operator for the normal samples.
+
+### Takeaway
+
+In this section, you've learned:
+
+- **Filtering data**: How to filter data with `filter`
+- **Splitting data**: How to split data into different channels based on a condition
+- **Viewing data**: How to use `view` to print the data
+
+We've now separated out the normal and tumor samples into two different channels. Next, we'll join the normal and tumor samples on the `id` field.
+
+---
+
+## 3. Join on sample ID
+
+In the previous section, we separated out the normal and tumor samples into two different channels. These could be processed independently using specific processes or workflows based on their type. But what happens when we want to compare the normal and tumor samples from the same patient? At this point, we need to join them back together making sure to match the samples based on their `id` field.
+
+Nextflow includes many methods for combining channels, but in this case the most appropriate operator is [`join`](https://www.nextflow.io/docs/latest/operator.html#join). If you are familiar with SQL, it acts like the `JOIN` operation, where we specify the key to join on and the type of join to perform.
+
+### 3.1. Use `map` and `join` to combine based on sample ID
+
+If we check the [`join`](https://www.nextflow.io/docs/latest/operator.html#join) documentation, we can see that it joins two channels based on the first item in each tuple. If you don't have the console output still available, let's run the pipeline to check our data structure and see how we need to modify it to join on the `id` field.
+
+```bash title="View normal and tumor samples"
+nextflow run main.nf
+```
+
+```console title="View normal and tumor samples"
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `main.nf` [loving_bardeen] DSL2 - revision: 012d38e59f
+
+[id:sampleA, repeat:1, type:tumor, bam:sampleA_rep1_tumor.bam]
+[id:sampleA, repeat:2, type:tumor, bam:sampleB_rep1_tumor.bam]
+[id:sampleB, repeat:1, type:tumor, bam:sampleC_rep1_tumor.bam]
+[id:sampleC, repeat:1, type:tumor, bam:sampleD_rep1_tumor.bam]
+```
+
+We can see that the `id` field is the first element in each map. For `join` to work, we should isolate the `id` field in each tuple. After that, we can simply use the `join` operator to combine the two channels.
+
+To isolate the `id` field, we can use the [`map` operator](https://www.nextflow.io/docs/latest/operator.html#map) to create a new tuple with the `id` field as the first element.
+
+=== "After"
+
+    ```groovy title="main.nf" linenums="2" hl_lines="5 8 9"
+    ch_samplesheet = Channel.fromPath("./data/samplesheet.csv")
+                        .splitCsv(header: true)
```

Comment:
Is it worth mentioning that we could have done the mapping here with the same results and that we do this here for teaching purposes?

Reply:
I am good with that. Might be something that people ask about and that the instructor can chat about live if necessary I think

Reply:
Do you want to put in an issue for this to be done as a followup? Merge requires that all comments be resolved and I agree this doesn't need to be done now but it seems like a shame to lose this.

Reply:
#606